### PR TITLE
backport: Replace fast sync with batch processing

### PR DESCRIPTION
**Pull request by gitferry**
Notable changes:
* remove fast sync completely
* introduce batch process for catching up
* introduce signature submission interval to config
* introduce batch submission size to config

### CODE REVIEW

**Review by SebastianElvis — APPROVED** [review]
Thanks! CI failed though

**Comment by gitferry** [comment]
@bap2pecs Do you have any ideas why op_e2e_test failed?

**Comment by bap2pecs** [comment]
> @bap2pecs Do you have any ideas why op_e2e_test failed?

i think it's due to some of our test cases depend on the fast sync logic. it probably need some non-trivial refactoring. maybe we can just disable the op_e2e_test in CI for now?

File: clientcontroller/babylon/babylon.go [hunk 1]
```diff
@@ -145,7 +146,125 @@ func (bc *BabylonController) RegisterFinalityProvider(
...
+
+	if res == nil {
+		return &types.TxResponse{}, nil
+	}
+
+	return &types.TxResponse{TxHash: res.TxHash}, nil
+}
+
+// UnjailFinalityProvider sends an unjail transaction to the consumer chain
+func (bc *BabylonController) UnjailFinalityProvider(fpPk *btcec.PublicKey) (*types.TxResponse, error) {
```
Comment (bap2pecs) [inline clientcontroller/babylon/babylon.go:246]: why is this added in this PR?

**Review by bap2pecs — APPROVED** [review]

File: clientcontroller/babylon/babylon.go (hunk 1 above)
Comment (gitferry) [reply clientcontroller/babylon/babylon.go:246]: Opps, should remove it

**Comment by gitferry** [comment]
> i think it's due to some of our test cases depend on the fast sync logic. it probably need some non-trivial refactoring. maybe we can just disable the op_e2e_test in CI for now?

Good. Removed it and added todos in ci

File: finality-provider/config/config.go [hunk 2]
```diff
@@ -55,21 +54,21 @@ var (
...
-	FastSyncLimit            uint64        `long:"fastsynclimit" description:"The maximum number of blocks to catch up for each fast sync"`
-	FastSyncGap              uint64        `long:"fastsyncgap" description:"The block gap that will trigger the fast sync"`
-	EOTSManagerAddress       string        `long:"eotsmanageraddress" description:"The address of the remote EOTS manager; Empty if the EOTS manager is running locally"`
-	MaxNumFinalityProviders  uint32        `long:"maxnumfinalityproviders" description:"The maximum number of finality-provider instances running concurrently within the daemon"`
-	SyncFpStatusInterval     time.Duration `long:"syncfpstatusinterval" description:"The duration of time that it should sync FP status with the client blockchain"`
+	LogLevel string `long:"loglevel" description:"Logging level for all subsystems" choice:"trace" choice:"debug" choice:"info" choice:"warn" choice:"error" choice:"fatal"`
+	// ChainType and ChainID (if any) of the chain config identify a consumer chain
+	ChainType                   string        `long:"chaintype" description:"the type of the consumer chain" choice:"babylon"`
```
Comment (bap2pecs) [inline finality-provider/config/config.go:59]: ah it's caused this line change

``
Whoops. There was an error while executing your fpd CLI 'failed to load configuration: /home/finality-provider/.fpd/fpd.conf:6: Invalid value `OPStackL2' for option `--chaintype'. Allowed values are: 'bash-5.1$ fpd commit-pubrand -h
```

File: finality-provider/service/fp_instance.go [hunk 3]
```diff
@@ -530,55 +393,52 @@ func (fp *FinalityProviderInstance) reportCriticalErr(err error) {
...
 	// error will be returned if maximum retries have been reached or the query to the consumer chain fails
 	for {
-		// error will be returned if max retries have been reached
-		var res *types.TxResponse
-		var err error
-		if len(targetBlocks) == 1 {
-			res, err = fp.SubmitFinalitySignature(targetBlocks[0])
-		} else {
+		select {
+		case <-time.After(fp.cfg.SubmissionRetryInterval):
```
Comment (bap2pecs) [inline finality-provider/service/fp_instance.go:410]: ok I just found another problem. this will wait `SubmissionRetryInterval` even at the first iteration. Default value is set to 300 seconds. that's why I found out batch sig submission is very slow

I will send out a fix soon

Comment (bap2pecs) [reply finality-provider/service/fp_instance.go:410]: https://github.com/babylonlabs-io/finality-provider/pull/217
